pkg/common/aws/ec2.go

```diff
@@ -31,3 +33,106 @@ func (CcsAwsSession *ccsAwsSession) CheckIfEC2ExistBasedOnNodeName(nodeName stri
 
 	return false, nil
 }
+
+// getSecurityGroups gets all security groups based on the security groups input struct
+func (CcsAwsSession *ccsAwsSession) getSecurityGroups(securityGroupsInput ec2.DescribeSecurityGroupsInput) (*ec2.DescribeSecurityGroupsOutput, error) {
+	err := CcsAwsSession.GetAWSSessions()
+	if err != nil {
+		return nil, err
+	}
+
+	securityGroups, err := CcsAwsSession.ec2.DescribeSecurityGroups(&securityGroupsInput)
+	if err != nil {
+		return nil, fmt.Errorf("error attempting to fetch security groups: %w", err)
+	}
+
+	if len(securityGroups.SecurityGroups) == 0 {
+		return nil, fmt.Errorf("no security groups found")
+	}
+
+	return securityGroups, nil
```

Comment:
nit: return a list of SGs instead of the wrapper response object

Reply:
Resolved in https://github.com/openshift/osde2e/pull/1558/commits/0de34542c4d705ab0513096a30d520fbf86d25bb